scipy/signal/_ltisys.py

```diff
@@ -1929,7 +1945,7 @@ def lsim2(system, U=None, T=None, X0=None, **kwargs):
 
         def fprime(x, t, sys, ufunc):
             """The vector field of the linear system."""
-            return dot(sys.A, x) + squeeze(dot(sys.B, nan_to_num(ufunc([t]))))
+            return dot(sys.A, x) + squeeze(dot(sys.B, nan_to_num(ufunc(t))))
```

Comment:
Why is this change necessary?

Reply:
Without this change `test_two_states` and `test_miso` fails.

"ValueError: shapes (2,2) and (1,2) not aligned: 2 (dim 1) != 1 (dim 0)" in `dot` line 1946

Reply:
Did this test not pass before? I would like to be clear on what make this change necessary before we proceed

Reply:
The original `test_miso` passed a 1d array for `u` into lsim, although the system consists of two inputs. I changed the test to be compliant to the documentation, then the test failed.